server/src/main/java/org/opensearch/common/time/JavaDateFormatter.java

```diff
@@ -318,9 +334,9 @@ public DateFormatter withZone(ZoneId zoneId) {
         if (zoneId.equals(zone())) {
             return this;
         }
-        List<DateTimeFormatter> parsers = new CopyOnWriteArrayList<>(
-            this.parsers.stream().map(p -> p.withZone(zoneId)).collect(Collectors.toList())
-        );
+        List<DateTimeFormatter> parsers = this.parsers.stream()
```

Comment:
Ah ... this is great finding, could you please change it to:

```
List<DateTimeFormatter> parsers = this.parsers.stream().map(p -> p.withZone(zoneId)).collect(Collectors.toList());
```

For the background, one of the parser caching optimization used to modify parsers list but it was not working well. This is sadly leftovers, the `CopyOnWriteArrayList` is not needed anymore, thank you.